pkg/pod/pod_test.go

```diff
@@ -2006,6 +2006,219 @@ debug-fail-continue-heredoc-randomly-generated-mz4c7
 	}
 }
 
+func TestPodBuild_TaskLevelResourceRequirements(t *testing.T) {
+	testcases := []struct {
+		desc                     string
+		ts                       v1beta1.TaskSpec
+		trs                      v1beta1.TaskRunSpec
+		expectedComputeResources []corev1.ResourceRequirements
+	}{{
+		desc: "only with requests",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}, {
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}},
+	}, {
+		desc: "only with limits",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+		}, {
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+		}},
+	}, {
+		desc: "both with requests and limits",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+				Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}, {
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}},
+	}, {
+		desc: "overwrite StepTemplate resources requirements",
```

Comment:
I think this is a mistake in the TEP :( 

Here, this says it's invalid: https://github.com/tektoncd/community/blob/main/teps/0104-tasklevel-resource-requirements.md#interaction-with-step-resource-requirements

However this example uses it: https://github.com/tektoncd/community/blob/main/teps/0104-tasklevel-resource-requirements.md#example-with-steptemplate

Would you be comfortable taking on a PR to the TEP fixing this mistake?

Reply:
got this. think we still will be good for overwriting resource requirements from StepTemplate while validating/avoiding ones from Step? 

yap. I can do this. (will raise today)

Reply:
Awesome thank you! I think if you can't specify compute resources on a step and a taskrun, you also shouldn't be able to do it on a steptemplate and a taskrun.

Reply:
right! I got your whole point now - as stepTemplate is for step.. 
thanks Lee, will mark it invalid - raise TEP update PR - then update code

Reply:
raised update PR https://github.com/tektoncd/community/pull/763

re-arranged validation in https://github.com/tektoncd/pipeline/pull/5054
and removed the StepTemplate case for pod_test.go